### PR TITLE
milestonr7: Validations and tests

### PR DESCRIPTION
**Pull request by abdessalem1998**
hello reviewer
In this pull request, I need to review my code for adding validations, model specs, and solving n+1 problems for the Blog app. in this part I did the following :

- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 RSpec.describe 'Posts', type: :request do
   describe 'GET #index' do
     before(:each) do
-      get user_posts_path(user_id: 1)
+      get user_posts_path(user_id: 3)
```
Comment (DeliceLydia) [inline spec/requests/posts_spec.rb:6]: - [ ] Please fix 2tests that are failing
fail

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @abdessalem1998, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Kudos on:
- An awesome README
- Solving the N+1 problem
- Following git workflow

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (abdessalem1998) [reply spec/requests/posts_spec.rb:6]: this happens because am using my data change with the data that you insert

Comment (abdessalem1998) [reply spec/requests/posts_spec.rb:6]: or run the seed file
will give you user id :1 
and post id :1

**Comment by arslanbisharat** [comment]
Hi @abdessalem1998,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**